e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+
+      var foundTheMessage = false;
+      deviceClient.open(function (openErr) {
+        debug('device has opened.');
+        if (openErr) {
+          done(openErr);
+        } else {
+          debug('about to connect a listener.');
+          deviceClient.on('message', function (msg) {
+            debug('received a message');
+            //
+            // Make sure that the message we are looking at is one of the messages that we just sent.
+            //
+            if (msg.messageId === uuidData) {
+              foundTheMessage = true;
+            }
+            //
+            // It doesn't matter whether this was a message we want, complete it so that the message queue stays clean.
+            //
+            deviceClient.complete(msg, function (err, result) {
+              if (err) {
+                done(err);
+              } else {
+                assert.equal(result.constructor.name, 'MessageCompleted');
+                if (foundTheMessage) {
+                  deviceClient.removeAllListeners('message');
+                  testRendezvous.imDone(deviceClientParticipant);
+                }
+              }
+            });
+          });
+          debug('about to open the service client');
+          serviceClient.open(function (serviceErr) {
+            debug('At service client open callback - error is:' + serviceErr);
+            if (serviceErr) {
+              done(serviceErr);
+            } else {
+              testRendezvous.imIn(serviceClientParticipant);
+              serviceClient.send(provisionedDevice.deviceId, message, function (sendErr) {
+                debug('At service client send callback - error is: ' + sendErr);
+                if (sendErr) {
+                  done(sendErr);
+                } else {
+                  testRendezvous.imDone(serviceClientParticipant);
+                }
+              });
+            }
+          });
+        }
+      });
+    });
+  });
+
+  describe('Over ' + deviceTransport.name + ' using device/eventhub clients - messaging', function () {
+    this.timeout(18000);
+
+    var deviceClient, ehClient, ehReceivers, provisionedDevice;
+
+    before(function (beforeCallback) {
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      ehClient = eventHubClient.fromConnectionString(hubConnectionString);
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+      ehReceivers = [];
+    });
+
+    afterEach(function (done) {
+      closeDeviceEventHubClients(deviceClient, ehClient, ehReceivers, done);
+    });
+    it('Device sends a message of zero size and it is received by the service', function (done) {
```

Comment:
nit: missing blank line i believe

Reply:
Fixed.

---
In reply to: [227564898](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227564898) [](ancestors = 227564898)